compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -189,12 +189,15 @@ object messages {
     val msg = {
       val ofFun =
         if (MethodType.syntheticParamNames(args.length + 1) contains param.name)
-          i" of expanded function $tree"
+          i" of expanded function:\n$tree"
         else
           ""
 
-      i"""missing parameter type for parameter ${param.name}$ofFun, expected = $pt
-         |The argument types of an anonymous function must be fully known. (SLS 8.5)"""
+      i"""missing parameter type
+         |
+         |The argument types of an anonymous function must be fully known. (SLS 8.5)
```

Comment:
It's better to have this general reference at the end, and put more specific information before.

Reply:
This is part of the `missing parameter type` explaination.

Reply:
It also follows the old format in scalac that users said was clearer.